FraihmworkHealthAndIntegrityApi.yaml

```
@@ -1787,6 +2020,71 @@ components:
         - sourceId
         - description
         - timeOfBlocklist
+           
+    UnavailabilitySchedule:
+      type: object
+      required:
+        - componentId
+        - schedules
+      properties:
+        componentId:
+          type: string
+          format: uuid
+          description: |
+            The unique identifier for the component. 
+            - This can represent one or more component IDs when the schedule is applied to multiple components.
+            - In cases where multiple component IDs are used, each component's unavailability schedule should be represented as a separate object within an array.
+        schedules:
+          type: array
+          items:
+            type: object
+            required:
+              - days
+              - timePeriods
+            properties:
+              days:
+                type: array
+                items:
+                  type: string
+                  enum:
+                    - Monday
+                    - Tuesday
+                    - Wednesday
+                    - Thursday
+                    - Friday
+                    - Saturday
+                    - Sunday
+                    - All
+                description: |
+                  Specifies the days of the week when the component is unavailable.
```

```suggestion
                  Specifies the days of the week that the time periods apply to
```
This object is used for POST, GET, and POST (remove), so it should be more generally worded

---

Changed description as suggested